viz/pkg/healthcheck/healthcheck.go

```diff
@@ -381,3 +388,62 @@ func fetchTapCaBundle(ctx context.Context, kubeAPI *k8s.KubernetesAPI) ([]*x509.
 	}
 	return caBundle, nil
 }
+
+func (hc *HealthChecker) checkPromAuthorized(ctx context.Context) error {
+	nses, err := hc.getDataPlaneNamespaces(ctx)
+	if err != nil {
+		return err
+	}
+
+	unauthorizedPods := []string{}
+	for _, ns := range nses {
+		// TODO(eliza): check if this namespace has an allow-scrapes config once
+		// that's implemented; if it has one, skip checking its pods.
+		pods, err := hc.KubeAPIClient().GetPodsByNamespace(ctx, ns.GetName())
+		if err != nil {
+			return fmt.Errorf("could not list pods in the %s namespace: %w", ns.GetName(), err)
+		}
+
+		var nsPrefix string
+		if ns.GetName() == hc.DataPlaneNamespace {
+			// if we're only checking one namespace, don't bother appending the
+			// namespace name to the pod's name in the error output
+			nsPrefix = ""
+		} else {
+			// otherwise, include the namespace name as well as the pod's
+			// name, since we are checking all namespaces.
+			nsPrefix = ns.GetName() + "/"
+		}
+
+		for _, pod := range pods {
+			defaultPolicy := pod.GetAnnotations()[k8s.ProxyDefaultInboundPolicyAnnotation]
```

Comment:
this is, perhaps, too pedantic, but technically this will fail if an uninjected pod has a default deny annotation on it for some reason.  We maybe don't care, but if we wanted to be totally correct we could use the `linkerd.io/control-plane-ns` label expression when listing pods in the namespace to only select injected pods.